patina_dxe_core/src/gcd/spin_locked_gcd.rs

```diff
@@ -2415,6 +2415,22 @@ impl SpinLockedGcd {
         result
     }
 
+    // Apply Patina memory protection rules to the given attributes.
+    //
+    // Rules:
+    //   - If UC memory is requested, always set XP memory.
+    fn apply_memory_protection_rules(attributes: u64) -> u64 {
+        let mut new_attributes = attributes;
+
+        // If we have UC memory, we always set XP. In DXE, we should never be executing from UC memory. On
+        // AArch64, this is defined as a programming error to have executable device memory (which UC maps to).
+        if attributes & efi::MEMORY_UC == efi::MEMORY_UC {
```

Comment:
Do you suspect there will be many more rules? If so, a more generic interface may be better:

``` rust
trait MemoryProtectionRule {
    fn apply(attributes: &mut u64)
}

/// A MemoryProtectionRule that will set XP memory if UC memory is requested
pub struct UcSetXp;

impl MemoryProtectionRule for UcSetXp {
    fn apply(attributes: &mut attributes) {
        if attributes & efi::MEMORY_UC == efi::MEMORY_UC {
            attributes |= efi::MEMORY_XP
        }
    }
}

fn apply_memory_protection_rules(mut attributes: u64) -> u64 {
    UcSetXp::apply(&mut attributes);
    AnotherRule::apply(&mut attributes);
    AndAnother::apply(&mut attributes);
    attributes
}
```

Reply:
At this point, no. There was a request to have this in a separate function for easy auditability, hence having it here. I will probably rearchitect this at some point because there are other places we apply policy, but it is not trivial to colocate them, so I need to think about how to do that more.